exporters/tests/test_pelorus_webhook_handler.py

```diff
@@ -244,3 +247,162 @@ async def test_pelorus_receive_pelorus_payload_error(headers, json_payload):
+            {"data" :"value", "data2" :"value2"}, # noqa
+        ),
+        (
+            b"My Secret",
+            "sha256=8b1e67669068057d751c49ca128fcf3b112dad8f26d2d00ece451c51b18351c1",
+            { "data": "value", "data2": "value2" }, # noqa
+        ),
+        (
+            b"My Secret",
+            "sha256=8b1e67669068057d751c49ca128fcf3b112dad8f26d2d00ece451c51b18351c1",
+            { "data":"value","data2":"value2" }, # noqa
+        ),
+        (
+            b"My Secret",
+            "sha256=8b1e67669068057d751c49ca128fcf3b112dad8f26d2d00ece451c51b18351c1",
+            { "data":"value", "data2":"value2" }, # noqa
+        ),
+        (
+            b"My Secret",
+            "sha256=8b1e67669068057d751c49ca128fcf3b112dad8f26d2d00ece451c51b18351c1",
+            { "data" :"value","data2" :"value2" }, # noqa
+        ),
+        (
+            b"My Secret",
+            "sha256=8b1e67669068057d751c49ca128fcf3b112dad8f26d2d00ece451c51b18351c1",
+            { "data" :"value", "data2" :"value2" }, # noqa
+        ),
+    ],
+)
+# fmt: on
+def test_verify_payload_signature_different_dict(
+    secret, expected_signature, json_payload_data
+):
+    """
+    Verifies if the json payload was properly verified based on provided hash.
+
+    Provided hash is calculated based on the exact payload information:
+        # Secret token = 'My Secret'
+        # Payload data = '{"data": "value", "data2": "value2"}'
+
+        # Calculated hash = sha256=8b1e67669068057d751c49ca128fcf3b112dad8f26d2d00ece451c51b18351c1
+
+    Even if the hash of other payload formats is different we should still find a match as the
+    format does not really affect it's content.
+    """
+    assert (
+        _verify_payload_signature(secret, expected_signature, json_payload_data) is True
+    )
+
+
+# fmt: off
+@pytest.mark.parametrize(
+    "secret,expected_signature,json_payload_data",
+    [
+        (
+            b"My Secret",
+            "sha256=f1dbf8a5d2aa74fb479c6bab52d80e947c66c98c131bb2fcfe97a6912623b05d",
+            {"data": "value", "data2": "value2"}, # noqa
+        ),
+        (
+            b"My Secret",
+            "sha256=ef6f04d63748a9d2809d45543a6badc26bc8198d4fc2d2a3338a8d7ce591d814",
+            {"data":"value","data2":"value2"}, # noqa
+        ),
+        (
+            b"My Secret",
+            "sha256=d5c59f2eca3b1ac55b1db4a8db2eff6439204d8687ea9fdd361347ead3fd4c2e",
+            { "data":"value","data2":"value2" }, # noqa
+        ),
+        (
+            b"My Secret",
+            "sha256=f1dbf8a5d2aa74fb479c6bab52d80e947c66c98c131bb2fcfe97a6912623b05d",
+            { "data":"value","data2":"value2" }, # noqa
+        ),
+        (
+            b"My Secret",
+            "sha256=2f6e3dde9a092d1fbce03bfbb0cb5d0b1db4419ab1e525a99684fd75b9ec8abb",
+            { "data":"value","data2":"value2" }, # noqa
+        ),
+    ],
+)
+# fmt: on
+def test_verify_payload_signature_different_hash(
```

Comment:
what this test does that the previous does not?

Reply:
The tests are very similar, however this one exercises different hashes instead of one. Different hashes mapping to different dictionaries formats. If you look closely at the comment of this test you can see that the hashes were calculated using 2 different methods on 2 different formats. This matches our documentation and will catch eventual issues.

In theory the previous test should be sufficient but we have 1-n mapping there, where here we have a better matrix.

Reply:
Also additional note here. We catch additional cases where the hash was calculated using shell method that produces the newline at the end of the file `echo {}` without `-n` flag. This is not covered by previous test.